src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -365,16 +365,20 @@ internal void AssertOrchestratorExists(string name, string version)
             }
         }
 
-        internal void AssertActivityExists(string name, string version)
+        internal bool TryGetActivityFunction(string name, string version, out string activityFunctions)
```

Comment:
>TryGetActivityFunction [](start = 22, length = 22)

I know I recommended you use this pattern, but as I look at it implemented I'm realizing that it's problematic because it breaks the convention of existing TryGet* methods which take a key and return a value. This signature is especially strange because you try to get a function and it returns a string list of all functions.

I feel really silly now...but I'm realizing that your original design was more clean and simple, even though it introduced a new type. Really sorry for the back and forth on this, but could we go back to that one (or a variation of it)?